### PR TITLE
Assignment 9 Doug Doner

### PR DESCRIPTION
**Pull request by dougdoner**

### CODE REVIEW

File: ShoppingList/ShoppingList/ShoppingItem.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ShoppingList
+{
+    public class ShoppingItem
+    {
+        public string ItemName { get; set; }
+
```
Comment (dsergio) [inline ShoppingList/ShoppingList/ShoppingItem.cs]: you could add other properties here such as quantity

File: ShoppingList/ShoppingList/MainWindowViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    public class MainWindowViewModel : BaseViewModel
+    {
+        public ObservableCollection<ShoppingItem> ShoppingList { get; } = new ObservableCollection<ShoppingItem>();
+    
+    
+        public MainWindowViewModel()
+        {
+            ShoppingList.Add(new ShoppingItem("Mayo"));
+            ShoppingList.Add(new ShoppingItem("Ketchup"));
+            ShoppingList.Add(new ShoppingItem("Red tomatoes"));
```
Comment (dsergio) [inline ShoppingList/ShoppingList/MainWindowViewModel.cs]: red tomatoes?

File: ShoppingList/ShoppingList/BaseViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace ShoppingList
+{
+    public class BaseViewModel : INotifyPropertyChanged
+    {
+        public event PropertyChangedEventHandler? PropertyChanged;
+
+        protected void SetProperty<T>(ref T field, T value,
```
Comment (dsergio) [inline ShoppingList/ShoppingList/BaseViewModel.cs]: looks good

File: ShoppingList/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<Project>
+ <PropertyGroup>
+   <Deterministic>true</Deterministic>
+   <TargetFramework>netcoreapp3.0</TargetFramework>
+   <Nullable>enable</Nullable>
```
Comment (dsergio) [inline ShoppingList/Directory.Build.props]: nice

File: ShoppingList/ShoppingList/ShoppingItem.cs (hunk 1 above)
Comment (dougdoner) [reply ShoppingList/ShoppingList/ShoppingItem.cs]: Oh yeah good idea thanks.

File: ShoppingList/ShoppingList/MainWindowViewModel.cs (hunk 2 above)
Comment (dougdoner) [reply ShoppingList/ShoppingList/MainWindowViewModel.cs]: Haha it was just for intial values.

File: ShoppingList/ShoppingList/MainWindowViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            ShoppingList.Add(new ShoppingItem("Eggs"));
+
+            SelectedItem = ShoppingList.First();
+
+            AddShoppingItemCommand = new Command(OnAddShoppingItem);
+        }
+
+        private void OnAddShoppingItem()
+        {
+            if(!(NewShoppingItem.Length == 0))
```
Comment (markdietzler) [inline ShoppingList/ShoppingList/MainWindowViewModel.cs]: You should run more checks here, break this out into a private helper class and run some more checks, like if it is white space, empty, or null.
`private bool HasNewShoppingItemChanged() {

//check for IsNullOrEmpty

//check for IsNullOrWhiteSpace

} `

Comment (dougdoner) [reply ShoppingList/ShoppingList/MainWindowViewModel.cs]: Oh okay thanks for the comment.

Comment (dougdoner) [reply ShoppingList/ShoppingList/MainWindowViewModel.cs]: Added some more tests in 98b9139

Comment (markdietzler) [reply ShoppingList/ShoppingList/MainWindowViewModel.cs]: No worries. :)

File: ShoppingList/ShoppingList/MainWindow.xaml [hunk 6]
```diff
@@ -5,8 +5,48 @@
...
+            <RowDefinition />
+        </Grid.RowDefinitions>
+
+        
+        <StackPanel Grid.Row="1"  Grid.Column="1" Background="AliceBlue">
+            <TextBlock Text="Edit Item"></TextBlock>
+            <TextBox Text="{Binding SelectedItem.ItemName, UpdateSourceTrigger=PropertyChanged}"/>
+        </StackPanel>
+
+        <StackPanel Grid.Row="2" Background="Fuchsia">
```
Comment (Keboo) [inline ShoppingList/ShoppingList/MainWindow.xaml]: Lovely choice of color.

File: ShoppingList/ShoppingList/MainWindow.xaml [hunk 7]
```diff
@@ -5,8 +5,48 @@
...
+            <ColumnDefinition/>
+        </Grid.ColumnDefinitions>
+
+        <Grid.RowDefinitions>
+            <RowDefinition />
+            <RowDefinition />
+        </Grid.RowDefinitions>
+
+        
+        <StackPanel Grid.Row="1"  Grid.Column="1" Background="AliceBlue">
```
Comment (Keboo) [inline ShoppingList/ShoppingList/MainWindow.xaml]: These controls are always visible. The intent was when no item is selected the view elements for editing an item should be hidden.

File: ShoppingList/ShoppingList/MainWindowViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            ShoppingList.Add(new ShoppingItem("Eggs"));
+
+            SelectedItem = ShoppingList.First();
+
+            AddShoppingItemCommand = new Command(OnAddShoppingItem);
+        }
+
+        private void OnAddShoppingItem()
+        {
+            if(!String.IsNullOrWhiteSpace(NewShoppingItem))
```
Comment (Keboo) [inline ShoppingList/ShoppingList/MainWindowViewModel.cs]:
```suggestion
            if(!string.IsNullOrWhiteSpace(NewShoppingItem))
```

**Pull request closed without merging**